### PR TITLE
Making Command's argument parsing easier.

### PR DESCRIPTION
**Pull request by MarioGK**
I made the process of parsing the arguments of the command a bit easier and modified the current commands to use it.

I hope @jhonynet will not mind me messing with his work.

### CODE REVIEW

**Comment by jhonynet** [comment]
@MarioGK no problem at all! i like your solution!

File: src/GameLogic/PlugIns/ChatCommands/TeleportChatCommandPlugIn.cs [hunk 1]
```diff
@@ -18,7 +19,7 @@ namespace MUnique.OpenMU.GameLogic.PlugIns.ChatCommands
     /// <seealso cref="MUnique.OpenMU.GameLogic.PlugIns.ChatCommands.IChatCommandPlugIn" />
     [Guid("ABFE2440-E765-4F17-A588-BD9AE3799886")]
     [PlugIn("Teleport chat command", "Handles the chat command '/teleport x y'. Teleports the character to the specified coordinates.")]
-    public class TeleportChatCommandPlugIn : IChatCommandPlugIn
+    public abstract class TeleportChatCommandPlugIn : IChatCommandPlugIn
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/TeleportChatCommandPlugIn.cs]: Is it really correct as 'abstract'?

Comment (MarioGK) [reply src/GameLogic/PlugIns/ChatCommands/TeleportChatCommandPlugIn.cs]: Nope, it was my mistake, fixed it

**Comment by MarioGK** [comment]
May i have the build error log? Locally it is building correctly

**Comment by sven-n** [comment]
```
PlugIns\ChatCommands\MoveChatCommandPlugIn.cs(20,42): error CS0535: 'MoveChatCommandPlugIn' does not implement interface member 'IChatCommandPlugIn.Usage' [D:\a\1\s\src\GameLogic\MUnique.OpenMU.GameLogic.csproj]
```

**Comment by MarioGK** [comment]
> ```
> PlugIns\ChatCommands\MoveChatCommandPlugIn.cs(20,42): error CS0535: 'MoveChatCommandPlugIn' does not implement interface member 'IChatCommandPlugIn.Usage' [D:\a\1\s\src\GameLogic\MUnique.OpenMU.GameLogic.csproj]
> ```

That file does not exists on my branch though

**Comment by sven-n** [comment]
It's really strange, I had something similar recently. When I merged it, it has broken the master build, and I had to delete an invisible file

File: src/GameLogic/PlugIns/ChatCommands/ArgumentsBase.cs [hunk 2]
```diff
@@ -16,16 +19,16 @@ public override string ToString()
...
             var stringBuilder = new StringBuilder();
-
+            bool isFirst = true;
             foreach (var property in properties)
             {
-                stringBuilder.Append($"{property.Name}:{property.GetValue(this)}");
-
-                // TODO this could be better
-                if (property != properties.Last())
+                if (!isFirst)
```
Comment (MarioGK) [inline src/GameLogic/PlugIns/ChatCommands/ArgumentsBase.cs]: This is the wrong logic, it is used to not add a space at the end of the string not at the first argument

Comment (sven-n) [reply src/GameLogic/PlugIns/ChatCommands/ArgumentsBase.cs]: But isn't the result the same (arguments seperated by a space)? I just turned the logic around ;-)

**Pull request merged**